### PR TITLE
 RS-658: add automatic to schema implementation for butane generic dependent structs

### PR DESCRIPTION
**Pull request by joaommartins**
Adds utoipa's ToSchema implementation to ForeignKey and Many.

Functionally, we declare the schema components in utoipa's openapi attribute, including the Many<T> and ForeignKey<T> ones, leading to the incorporation of the respective T type in the OpenAPI spec.

### CODE REVIEW

File: butane_core/Cargo.toml [hunk 1]
```diff
@@ -52,3 +52,10 @@ fake = { version = "2.5.1-alpha.1", registry = "franklin-ai-rosalind", features
   "uuid",
 ] }
 rand = "0.8"
+utoipa = { version = "=3.0.4-alpha.1", registry = "franklin-ai-rosalind", features = [
+  "actix_extras",
```
Comment (jayvdb) [inline butane_core/Cargo.toml]: do we need all these features for butane ?

Comment (joaommartins) [reply butane_core/Cargo.toml]: Good point, all we need here is the ToSchema trait.

**Review by jayvdb — APPROVED** [review]

File: butane/Cargo.toml [hunk 2]
```diff
@@ -25,8 +25,8 @@ tls = ["butane_core/tls"]
 uuid = ["butane_core/uuid", "butane_codegen/uuid"]
 
 [dependencies]
-butane_codegen = { version = "0.5.1-alpha.4", registry = "franklin-ai-rosalind" }
-butane_core = { version = "0.5.2-alpha.3", registry = "franklin-ai-rosalind" }
+butane_codegen = { path = "../butane_codegen", version = "0.5.1-alpha.6", registry = "franklin-ai-rosalind" }
```
Comment (jayvdb) [inline butane/Cargo.toml:28]: remove these?

File: butane_cli/Cargo.toml [hunk 3]
```diff
@@ -18,7 +18,7 @@ sqlite-bundled = ["butane/sqlite-bundled"]
 
 [dependencies]
 anyhow = "1.0"
-butane = { version = "0.5.2-alpha.4", registry = "franklin-ai-rosalind", features = [
+butane = { path = "../butane", version = "0.5.2-alpha.6", registry = "franklin-ai-rosalind", features = [
```
Comment (jayvdb) [inline butane_cli/Cargo.toml:21]: again remove?

File: butane_codegen/Cargo.toml [hunk 4]
```diff
@@ -13,7 +13,7 @@ datetime = []
 
 [dependencies]
 proc-macro2 = { version = "1.0", default-features = false }
-butane_core = { version = "0.5.2-alpha.3", registry = "franklin-ai-rosalind" }
+butane_core = { path = "../butane_core", version = "0.5.2-alpha.5", registry = "franklin-ai-rosalind" }
```
Comment (jayvdb) [inline butane_codegen/Cargo.toml:16]: again

File: butane_core/Cargo.toml [hunk 5]
```diff
@@ -52,3 +52,4 @@ fake = { version = "2.5.1-alpha.1", registry = "franklin-ai-rosalind", features
   "uuid",
 ] }
 rand = "0.8"
+utoipa = { path = "../../utoipa/utoipa", version = "=3.0.4-alpha.2", registry = "franklin-ai-rosalind" }
```
Comment (jayvdb) [inline butane_core/Cargo.toml:55]: really remove the `path`

**Review by jayvdb — APPROVED** [review]
please merge so I can add stuff on top

**Pull request merged**